manifests/params.pp

```diff
@@ -6,14 +6,26 @@
 class consul::params {
   $manage_repo = false
 
-  case $facts['os']['architecture'] {
-    'x86_64', 'x64', 'amd64': { $arch = 'amd64' }
-    'i386':                   { $arch = '386' }
-    'aarch64':                { $arch = 'arm64' }
-    'armv7l':                 { $arch = 'armhfv6' }
-    /^arm.*/:                 { $arch = 'arm' }
-    default:                  {
-      fail("Unsupported kernel architecture: ${facts['os']['architecture']}")
+  if versioncmp ($consul::version, '1.9.11') >= 0 {
```

Comment:
this fails because the `consul::params` class doesn't know the `$consul::version` variable. I suggest to add a variable `$version` to theh `consul::params` class and use it in your new code. Then update the init.pp to use `$consul::params::version`.

Reply:
picking this up again - apologies for the delay